gflownet/envs/crystals.py

```diff
@@ -2,112 +2,118 @@
-        Minimum number of elements of each used kind that needs to be used to construct a crystal
-
-    max_atom_i : int
-        Maximum number of elements of each kind that can be used to construct a crystal
     """
 
     def __init__(
         self,
+        elements: Union[List, int] = 84,
         max_diff_elem: int = 4,
         min_diff_elem: int = 2,
-        periodic_table: int = 84,
-        oxidation_states: Optional[dict] = None,
-        alphabet: Optional[dict] = None,
         min_atoms: int = 2,
         max_atoms: int = 20,
         min_atom_i: int = 1,
         max_atom_i: int = 10,
-        env_id=None,
-        reward_beta=1,
-        reward_norm=1.0,
-        reward_norm_std_mult=0,
-        reward_func="power",
-        denorm_proxy=False,
-        energies_stats=None,
-        proxy=None,
-        oracle=None,
+        oxidation_states: Optional[Dict] = None,
+        alphabet: Optional[Dict] = None,
+        required_elements: Optional[Union[Tuple, List]] = (3,),
         **kwargs,
     ):
-        super(Crystal, self).__init__(
-            env_id,
-            reward_beta,
-            reward_norm,
-            reward_norm_std_mult,
-            reward_func,
-            energies_stats,
-            denorm_proxy,
-            proxy,
-            oracle,
-            **kwargs,
-        )
+        """
+        Attributes
+        ----------
+        elements : list or int
+            Elements that will be used for construction of crystal. Either list, in which case every value should
+            indicate the atomic number of an element, or int, in which case n consecutive atomic numbers will
+            be used. Note that we assume this will correspond to real atomic numbers, i.e. start from 1, not 0.
+
+        max_diff_elem : int
+            Maximum number of unique elements in the crystal
+
+        min_diff_elem : int
+            Minimum number of unique elements in the crystal
+
+        min_atoms : int
+            Minimum number of atoms that needs to be used to construct a crystal
+
+        max_atoms : int
+            Maximum number of atoms that can be used to construct a crystal
+
+        min_atom_i : int
+            Minimum number of elements of each used kind that needs to be used to construct a crystal
+
+        max_atom_i : int
+            Maximum number of elements of each kind that can be used to construct a crystal
+
+        oxidation_states : (optional) dict
+            Mapping from ints (representing elements) to lists of different oxidation states
 
-        self.state = [0 for _ in range(periodic_table)]  # atom counts for each element
+        alphabet : (optional) dict
+            Mapping from ints (representing elements) to strings containing human-readable elements' names
+
+        required_elements : (optional) list
+            List of elements that must be present in a crystal for it to represent a valid end state
+        """
+        super(Crystal, self).__init__(**kwargs)
```

Comment:
A question just for the sake of learning from you @michalkoziarski: I thought that in recent Python versions (from 3+?) `super()` is equivalent to `super(ClassName, self)`. Is there a good reason to keep using `super(ClassName, self)`? (is it just because "explicit is better than implicit¨?)

Reply:
Like we discussed :) but just for posterity, I don't see any reason other than backward compatibility to Python 2 (which doesn't matter in our case because the rest of code isn't compatible anyway), it was like that before and I didn't change it.